scripts/place_spread_orders.ts

```diff
@@ -0,0 +1,146 @@
+const BatchExchange = artifacts.require("BatchExchange");
+import { factory } from "../src/logging";
+const log = factory.getLogger("scripts.spread_orders");
+
+import BN from "bn.js";
+import readline from "readline";
+
+import { sendTxAndGetReturnValue } from "../test/utilities";
+import { fetchTokenInfoFromExchange } from "../src/fee_token_liquidity";
+import type { TokenInfo } from "../src/fee_token_liquidity";
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const promptUser = function (message: string): Promise<string> {
+  return new Promise((resolve) =>
+    rl.question(message, (answer) => resolve(answer)),
+  );
+};
+
+const formatAmount = function (amount: number, token: TokenInfo) {
+  const exponent = token.decimals;
+  if (exponent) {
+    return new BN(10).pow(new BN(exponent)).muln(amount);
+  }
+  throw new Error("Insufficient token data for amount formatting");
+};
+
+const argv = require("yargs")
+  .option("tokens", {
+    alias: "t",
+    type: "string",
+    describe: "Collection of trusted tokenIds",
+    coerce: (str: string) => {
+      return str.split(",").map((t) => parseInt(t));
+    },
+  })
+  .option("accountId", {
+    describe: "Account index of the order placer",
+    type: "int",
+    default: 0,
+  })
+  .option("spread", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.25,
+  })
+  .option("sellAmount", {
+    type: "int",
+    describe: "Maximum sell amount (in full token units)",
+    default: 1000,
+  })
+  .option("validFrom", {
+    type: "int",
+    describe: "Number of batches (from current) until order become valid",
+    default: 3,
+  })
+  .option("expiry", {
+    type: "int",
+    describe: "Maximum auction batch for which these orders are valid",
+    default: 2 ** 32 - 1,
+  })
+  .demand(["tokens"])
+  .help(
+    "Make sure that you have an RPC connection to the network in consideration. For network configurations, please see truffle-config.js Example usage \n   npx truffle exec scripts/place_spread_orders.js --tokens=2,3,4 --accountId 0 --spread 0.3 --validFrom 5",
+  )
+  .version(false).argv;
+
+module.exports = async (callback: Truffle.ScriptCallback) => {
+  try {
+    const instance = await BatchExchange.deployed();
+    const accounts = await web3.eth.getAccounts();
+    const account = accounts[argv.accountId];
+
+    const batch_index = (await instance.getCurrentBatchId()).toNumber();
+    const token_data = await fetchTokenInfoFromExchange(
```

Comment:
My suggestion would be to not make `src/fee_token_liquidity.ts` functions log. Instead, the callers (for example this script) should log the token information as appropriate.

This would also allow us to make the logger a `devDependency`.

Reply:
This is a cool idea, but there are like 3 or four others places with logs in this file. Should I just eliminate all of them?

Reply:
I'm gonna give it a try!

Reply:
If the logs are necessary, the caller should pass in the logger, so that it can have the same log tag.